flexget/plugins/clients/deluge.py

```diff
@@ -660,16 +660,20 @@ def rename(file, new_name):
                             opts.get('main_file_ratio') * 100))
 
                 if config.get('container_directory'):
-                    if len(status['files']) > 1:
-                        log.info('Renaming Folder %s to %s', status['name'], entry.render(config.get('container_directory')))
-                        main_file_dlist.append(client.core.rename_folder(torrent_id, status['name'], pathscrub(entry.render(config.get('container_directory')))))
+                    if opts.get('content_filename') or opts.get('main_file_only'):
+                        folder_structure = big_file_name.split(os.sep)
```

Comment:
This will throw the error ``local variable 'big_file_name' referenced before assignment`` if ``content_filename`` or ``main_file_only`` are set but there isn't one file above the "big" (main) file threshold, or if ``content_filename`` is not set ([line 626](https://github.com/Flexget/Flexget/pull/1790/files/197a4ed67009244f4c1d62fcb652d1b7b319cb20..35106e05fa88e48ca6c5d70012669b3c89bb8245#diff-a9b84b3d1f44f7e89a9560144236437eR626)). I'd add ``big_file_name = ''`` above [line 562](https://github.com/Flexget/Flexget/pull/1790/files/197a4ed67009244f4c1d62fcb652d1b7b319cb20..35106e05fa88e48ca6c5d70012669b3c89bb8245#diff-a9b84b3d1f44f7e89a9560144236437eR562) to ensure it's always assigned to something, and then just do ``if big_file_name`` here.